src/tiledb/cloud/utilities/_common.py

```diff
@@ -347,3 +382,94 @@ def wrapper(*args, **kwargs) -> Dict[str, object]:
         return {"status": "started", "graph_id": str(graph.server_graph_uuid)}
 
     return wrapper
+
+
+def find(
+    uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    include: Optional[Union[str, Callable]] = None,
+    exclude: Optional[Union[str, Callable]] = None,
+    max_count: Optional[int] = None,
+) -> Iterator[str]:
+    """Searches a path for files matching the include/exclude pattern using VFS.
+
+    :param uri: Input path to search
+    :param config: Optional dict configuration to pass on tiledb.VFS
+    :param include: Optional include pattern string
+    :param exclude: Optional exclude pattern string
+    :param max_count: Optional stop point when searching for files
+    :param count: Current number of files found
+    """
+
+    with tiledb.scope_ctx(config):
+        vfs = tiledb.VFS(config=config, ctx=tiledb.Ctx(config))
+        listing = vfs.ls(uri)
+        current_count = 0
+
+        def list_files(listing):
+            for f in listing:
+                # Avoid infinite recursion
+                if f == uri:
+                    continue
+
+                if vfs.is_dir(f):
+                    yield list_files(
```

Comment:
This should be `yield from` and I believe we want to call `find` not `list_files` to recurse into the sub-folders. This means we also need to pass all the parameters, `config=config, max_count=max_count`.

Reply:
I will add a test case. For `find` and not `list_files`, we retain the current count as a check in the outer function. With `list_files` I have removed the extraneous arguments.